lib/kernel/src/pg.erl

```diff
@@ -395,17 +395,13 @@ handle_info({leave, Peer, PidOrPids, Groups}, #state{scope = Scope, remote = Rem
     end;
 
 %% we're being discovered, let's exchange!
-handle_info({discover, Peer}, #state{remote = Remote, local = Local} = State) ->
-    gen_server:cast(Peer, {sync, self(), all_local_pids(Local)}),
-    %% do we know who is looking for us?
-    case maps:is_key(Peer, Remote) of
-        true ->
-            {noreply, State};
-        false ->
-            MRef = erlang:monitor(process, Peer),
-            erlang:send(Peer, {discover, self()}, [noconnect]),
-            {noreply, State#state{remote = Remote#{Peer => {MRef, #{}}}}}
-    end;
+handle_info({discover, Peer}, State) ->
+    handle_discover(Peer, State);
+
+%% New discover msg sent by a future pg version.
+%% Accepted first in OTP 26, to be used by OTP 28 or later.
+handle_info({discover, Peer, _ProtocolVersion}, State) ->
```

Comment:
`handle_info` spec is missing this clause (probably spec should be rewritten completely, using overloaded function specs).
With the spec, it'd also be clearer what is the type of _ProtocolVersion (I don't know whether it should be an integer, a bitmask, or a map of "capabilities" for easier expansion).

Reply:
I don't see much reason to specify the ProtocolVersion type now. I think it's better to leave that up to the author of a future protocol update.